### PR TITLE
NAS-119833 / 23.10 / Allow time slicing for NVIDIA gpu

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

A PR (https://github.com/truenas/middleware/pull/10411) was raised by @Ornias1993 to add timeslicing support recently added to nvidia device plugin.

This continues on those changes and adds support to allow using a single NVIDIA gpu in multiple apps defaulting to 5 apps similar to what we do for AMD.

### CODE REVIEW

**Comment by Ornias1993** [comment]
I want to note, that it's normal practice in opensource to use co-author tags on the commit and correctly name people in PR's, instead of refering to them as "community".

Considering iX has stated to really like more opensource contributions, I think it's good to stick to those practices.

File: src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py [hunk 1]
```diff
@@ -120,6 +128,29 @@
...
+GPU_CONFIGMAPS = {
+    'NVIDIA': {
+        'apiVersion': 'v1',
+        'kind': 'ConfigMap',
+        'metadata': {
+            'name': 'nvidia-device-plugin-config',
+            'namespace': 'kube-system'
+        },
+        'data': {
+            'nvdefault.yaml': """version: v1
```
Comment (Ornias1993) [inline src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py:141]: @stavros-k and I where not sure if we needed to add `|` here, which is required for multiline in kube yaml.

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py:141]:
```
Allocatable:
  cpu:                16
  ephemeral-storage:  612650699501
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             65804624Ki
  nvidia.com/gpu:     5
  pods:               250
  ```
  
GPU changes ^^^

Comment (Ornias1993) [reply src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py:141]: Ahh you mean you already tested? :)

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py:141]: Ofc ✅

Comment (Ornias1993) [reply src/middlewared/middlewared/plugins/kubernetes_linux/gpu.py:141]: I stand corrected, good to know python multiline syntax translates correctly to kube yaml...
Will make note for next time! :)

**Review by Ornias1993 — COMMENTED** [review]
Primary potential issue with this was potential yaml syntax translation for multi-line strings, which is tested to be working correctly.

Nice work @Qubad786 
LGTM.

**Review by yocalebo — APPROVED** [review]

**Comment by Ornias1993** [comment]
@Qubad786 can you backporg this for 22.12.2?

**Comment by Qubad786** [comment]
backport

**Pull request merged**